server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/routes/MailRepositoriesRoutesTest.java

```diff
@@ -465,6 +469,54 @@ public void retrievingAMailShouldFailWhenUnknown() throws Exception {
             .body("message", is("Could not retrieve " + name));
     }
 
+    @Test
+    public void downloadingAMailShouldReturnTheEml() throws Exception {
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecificationWithPortOnly(webAdminServer.getPort())
+                .setBasePath(MailRepositoriesRoutes.MAIL_REPOSITORIES)
+                .build();
+        RestAssured.registerParser(Constants.RFC822_CONTENT_TYPE, Parser.JSON);
+
+        when(mailRepositoryStore.get(URL_MY_REPO)).thenReturn(Optional.of(mailRepository));
+
+        String name = NAME_1;
+        FakeMail mail = FakeMail.builder()
+            .name(name)
+            .fileName("mail.eml")
+            .build();
+        mailRepository.store(mail);
+
+        String expectedContent = IOUtils.toString(mail.getMessage().getRawInputStream(), StandardCharsets.UTF_8);
```

Comment:
Sad... You don't really test things and fail to recognise you do return partial data.

Here you should read data from the file.